straxen/plugins/peaks/peak_se_score.py

```diff
@@ -0,0 +1,119 @@
+import numpy as np
+from numba import njit
+
+import strax
+import straxen
+
+
+class PeakSEScore(strax.OverlapWindowPlugin):
+    """This plugin is designed to calculate a score based on the position relations to single
+    electrons.
+
+    References:
+        * v0.2.0: xenon:xenonnt:ac:sr1:hotspot_veto_cut:wimp_roi
+        * position resolution: xenon:xenonnt:svetter:data_driven_resolution_sr1a0_vs_sr1a1
+
+    """
+
+    __version__ = "0.4.0"
+    depends_on = ("peak_basics", "peak_positions")
+    provides = "peak_se_score"
+    save_when = strax.SaveWhen.EXPLICIT
+
+    dtype = strax.time_fields + [
+        (
+            ("Sum of the PDF of the SE position nearby probability.", "se_score"),
+            np.float32,
+        ),
+    ]
+
+    se_time_search_window_left = straxen.URLConfig(
+        default=5e9, type=int, help="SEs time window searching window, left side [ns]"
+    )
+
+    se_time_search_window_right = straxen.URLConfig(
+        default=0,
+        type=int,
+        help="Time searching window right side extension, one max drift time [ns]",
+    )
+
+    def get_window_size(self):
+        # This method is required by the OverlapWindowPlugin class
+        return 2 * (self.se_time_search_window_left + self.se_time_search_window_right)
+
+    def setup(self):
+        self._para_a = 1
+        self._para_b = 1
+        self._sr_phase = 1
+
+    def select_se(self, peaks):
+        """Function which select single electrons from peaks.
+
+        :param peaks: peaks data contains single electrons.
+        :return: single electron peaks data
+
+        """
+        mask = peaks["type"] == 2
+        mask &= (peaks["area"] > 10) & (peaks["area"] < 80)
+        mask &= (peaks["range_50p_area"] > 80) & (peaks["range_50p_area"] < 700)
+        return mask
+
+    @staticmethod
+    @njit
+    def get_se_count_and_pdf_sum(peaks, se_peaks, se_indices, para_a, para_b, eps):
+        se_nearby_probability = np.zeros(len(peaks))
+        for index, peak_i in enumerate(peaks):
+            indices = se_indices[index]
+            se_in_time = se_peaks[indices[0] : indices[1]]
+            peak_area_top = peak_i["area"] * (peak_i["area_fraction_top"] + eps)
+            se_area_top = se_in_time["area"] * (se_in_time["area_fraction_top"] + eps)
```

Comment:
What is the purpose of `eps` here?

Reply:
Some peaks have `aft == 0` which will cause numba to error. So this way to avoid divided by zero.

Reply:
Will the pos-rec of the peaks with `area_fraction_top  == 0` be `np.nan`?

Reply:
Yes, the pos-rec gives these peaks `np.nan`. These peaks mainly `type==0`

Reply:
Yes, the pos-rec gives these peaks `np.nan`. These peaks mainly `type==0`

Reply:
Then is it better to replace the `eps` to mask about whether pos-rec is `np.nan`?

Reply:
No `eps` is not easily understood by people.

Reply:
Ok, let's do it by adding a comment to help the readability.

Reply:
I am pretty sure you should identify the `np.nan` and use a mask to make it clear. If you need help with how the mask works, @yuema137 and I can help.

Reply:
Could you be more explicit about the reason: )

Reply:
1. If a peak has position `np.nan`, the `se_score` of it should be `np.nan`, but in your calculation, it is a relatively small number but not `np.nan`.
2. It is not hard to just set the score of peaks with position `np.nan` to `np.nan`.
3. Though people can read the comments, they will also wonder why the developer does not set the score directly to `np.nan`.
4. The value of `eps` depends on a specific IEEE standard which is hard to track.

Reply:
Well as you insist, I'll modify it accordingly to proceed.